### DISCUSSION TITLE
Git sync with local repo? (Amazing work on this so far.)

### DISCUSSION THREAD

**Discussion by ReessKennedy** [redimp/otterwiki#87] [Ideas]
I just spun this puppy up using docker compose and love how simple and fast it is. I also love the right sidebar that lists the content of the page and the "All pages" index. 

All that said, I am really looking for a solution that I can use with a very small team of mine and some of use need to have a web editor and some of use can just edit markdown locally so my question: Can Otter be our hub by giving us a method of pulling and pushing to the wiki's git repo? 

if so, this would be huge. I was thinking I might find a link to a repo like if I were using Github's Gollum-based wiki or have some method to push/pull like with Gitbook or Ikiwiki but perhaps not? 

Maybe I am overlooking something though! Thanks for you work here @redimp.

**Comment by redimp** [comment]
Thanks for opening up this discussion. The feature that you can clone the wiki locally, work on it and push the changes is on my list for a long time.

Regardless of the fact that you might want to work locally, it seems to be important to have the information in the wiki available in different repositories. Pushing the data into remote repositories is (see #24) on my list, too.

I think it necessary to add that An Otter Wiki cares less about the file name case, as requested in #85, to make sure that anything created outside and pushed into the wikis repository get represented.

**Comment by redimp** [comment]
In #88 the config option `RETAIN_PAGE_NAME_CASE` was added (not in a release yet). This is an important prerequisite for enableing pushing/pulling from An Otter Wiki.

I have started to implement the feature of serving git via http. Proof of concept works, will now do the fine tuning.

As far as permissions are concerned, I think it makes sense to handle this in the same way as in the web interface: If you can read, you can pull. If you can write, you can push. For the write check the user must be allowed to manage attachments. The reason for this is that it is not possible to distinguish whether a page or an attachment is included in the push.

**Reply by ReessKennedy** [reply]
Awesome! Looking forward to trying. Thanks @redimp !

**Comment by redimp** [comment]
Feature released in [2.2.0](https://github.com/redimp/otterwiki/releases/tag/v2.2.0)

**Reply by ReessKennedy** [reply]
Wow! Will def check out Otterwiki again because of this new release.

**Reply by lewiswalsh** [reply]
If I add a remote to the local repo will OtterWiki push automatically, or do I need to set up a cronjob for this?

**Reply by redimp** [reply]
Hey @lewiswalsh, there is no automatism yet. And the wikican not push the repository yet (feature request in #24). Setting up a cronjob that pulls the wiki would be the solution I would choose.

**Reply by dcronin05** [reply]
I feel like I'm missing something really obvious given that I can't find it explained anywhere but *how* do I clone the repo? What is the address i'd use? AOW is hosted locally in a container at 10.0.0.59 on port 8099. I tried `git clone http://10.0.0.59:8099/repository` just to see but obviously it didn't work. What's the super simple answer to this that I missed?

**EDIT**: Well, apparently I was on the right track. It was `git clone http://10.0.0.59:8099/.git`

**Reply by redimp** [reply]
This is not well documented, my bad. You find the url in the users settings, if the user is allowed to clone the repository, e.g.:

<img width="523" alt="image" src="https://github.com/user-attachments/assets/1d9ed3ec-2240-4ab6-9475-d80c559a63a6">
